internal/server/openapi.go

```diff
@@ -146,17 +143,17 @@ func orderedTagNames() []string {
 	return tagNames
 }
 
-func createComponent(rst reflect.Type) *openapi3.SchemaRef {
-	if openAPISchema.Components.Schemas == nil {
-		openAPISchema.Components.Schemas = openapi3.Schemas{}
+func createComponent(openAPIDoc openapi3.T, rst reflect.Type) *openapi3.SchemaRef {
```

Comment:
are you sure this works? openapi3.T is a struct and will be copied when passed by value

Reply:
Great question! I missed this when I was adding the args.

This worked because we only ever modified the schemas map (which is always pass by reference), but it is still really surprising.

I pushed another commit which changes the first arg to a few functions to be just the `openapi.Schemas`, which is a `map[string]*SchemaRef`. That should prevent any surprises in the future because we only pass the mutable part to these functions.